### ISSUE TITLE
hostpython recipe fails on El Capitan and above due to LibreSSL

### ISSUE THREAD

**Issue by Frimkron** [kivy/kivy-ios#270]
When building kivy (with the openssl recipe, in case that makes a difference) I get a linker error for the hostpython recipe:

```
Undefined symbols for architecture x86_64:
  "_RAND_egd", referenced from:
      _PySSL_RAND_egd in libpython2.7.a(_ssl.o)
  "_SSLv2_method", referenced from:
      _PySSL_sslwrap in libpython2.7.a(_ssl.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [python.exe] Error 1
```

I'm using the master branch at the time of writing (535abf0e6b285fb5541492 from Aug 11 2017), XCode 8.2.1 and macOS 10.13.

As far as I've been able to figure out, it looks like it's linking to the system SSL library which as of macOS 10.11 is LibreSSL rather than OpenSSL, hence the missing symbols.

I was able to work around this by invoking the linker by hand and including the path to my brew-installed OpenSSL library (last argument here):

    $ clang -Qunused-arguments -fcolor-diagnostics -lsqlite3 -lffi \
    -L/Users/xyz/kivy-ios/dist/hostlibffi/usr/local/lib -o python.exe Modules/python.o libpython2.7.a \
    -ldl -framework CoreFoundation -L/Users/xyz/kivy-ios/dist/hostpython/lib -lz -lssl -lcrypto \
    -L/usr/local/opt/openssl/lib

**Comment by inclement** [comment]
Thanks for the report. I'm not actually sure what would cause this (we should have changed all the paths so it wouldn't find a system library), but hopefully it should be easy enough to fix.

However, I don't have any apple hardware to test with, so hopefully someone else will be able to look into it.

**Comment by cruor99** [comment]
I recently found another workaround for this issue, that may also be a simpler workaround. 
I first ran ./toolchain.py build hostpython
Then I ran ./toolchain.py build openssl kivy --arch=armv7 --arch=arm64

Then I went back up the app directory and ran buildozer ios release like normal, and it worked like a charm. 

The drawback to this, is that it won't run in the simulator (as that requires the x86_64 architecture).
